src/simsoptpp/biot_savart_vjp_impl.h

```diff
@@ -172,9 +175,112 @@ void biot_savart_vjp_kernel(AlignedPaddedVec& pointsx, AlignedPaddedVec& pointsy
     }
 }
 
+#else
+
+template<class T, int derivs>
+void biot_savart_vjp_kernel(AlignedPaddedVec& pointsx, AlignedPaddedVec& pointsy, AlignedPaddedVec& pointsz,
+            T& gamma, T& dgamma_by_dphi, T& v, T& res_gamma, T& res_dgamma_by_dphi, T& vgrad, T& res_grad_gamma,
+            T& res_grad_dgamma_by_dphi) {
+    if(gamma.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("gamma needs to be in row-major storage order");
+    if(dgamma_by_dphi.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("dgamma_by_dphi needs to be in row-major storage order");
+    if(res_gamma.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("res_gamma needs to be in row-major storage order");
+    if(res_dgamma_by_dphi.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("res_dgamma_by_dphi needs to be in row-major storage order");
+    if(res_grad_gamma.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("res_grad_gamma needs to be in row-major storage order");
+    if(res_grad_dgamma_by_dphi.layout() != xt::layout_type::row_major)
+          throw std::runtime_error("res_grad_dgamma_by_dphi needs to be in row-major storage order");
+    int num_points         = pointsx.size();
+    int num_quad_points    = gamma.shape(0);
+    constexpr int simd_size = 1;
```

Comment:
In the NON-XSIMD code, simd_size=1 implies no vectorization. Since I am using constexpr, it will be optimized away during compilation. Anyhow I deleted the variable now, but it shouldn't matter due to compile-time optimization of constexpr variables.